CHANGES.rst

```diff
@@ -1,16 +1,20 @@
 Changes
 =======
 
-Version 0.7.3 (UNRELEASED)
+Version 0.7.3 (2021-03-17)
 --------------------------
 
+- Users:
+    - Fixes problem with jobs stuck in `running` state when there are network issues, improving the overall platform resilience.
+    - Fixes problem related to Slurm job submissions.
```

Comment:
- Fixes job dispatch issues to HTCondor and Slurm compute backends.

(Although it would be good to say what kind of issues or problems... e.g. `Fixes job command encoding issues when dispatching jobs to HTCondor and Slurm backends.` or the like.)